### PR TITLE
Add Docker image build and push on release workflow

### PR DESCRIPTION
**Pull request by klejejs**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Build and push Stun server Docker image on release
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  deploy:
```
Comment (ludeeus) [inline .github/workflows/release.yml:9]: Guard this so it can only be started in this repository.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Extract release version
+        run: |
+          echo "RELEASE_VERSION=$(echo $GITHUB_REF | cut -d / -f 3)" >> $GITHUB_ENV
+
+      - name: Build and push
+        uses: docker/build-push-action@v6.7.0
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
```
Comment (ludeeus) [inline .github/workflows/release.yml]: Why are we building a multi arch image?
Is amd64 not enough?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract release version
+        run: |
+          echo "RELEASE_VERSION=$(echo $GITHUB_REF | cut -d / -f 3)" >> $GITHUB_ENV
```
Comment (ludeeus) [inline .github/workflows/release.yml]: The ref is not always present.
Use the tag form the release event, and add it to the step exports.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Build and push
+        uses: docker/build-push-action@v6.7.0
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ghcr.io/${{ github.repository }}:latest,ghcr.io/${{ github.repository }}:${{ env.RELEASE_VERSION }}
```
Comment (ludeeus) [inline .github/workflows/release.yml]: Remove "latest"

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'home-assistant/stun' }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.7
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3.2.0
```
Comment (ludeeus) [inline .github/workflows/release.yml]: Is this really needed for an amd64 only build?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v6.7.0
+        if: ${{ github.event_name == 'release' }}
+        with:
+          context: .
+          platforms: linux/amd64
+          push: true
```
Comment (ludeeus) [inline .github/workflows/release.yml]:
```suggestion
          push: ${{ github.event_name == 'release' }}
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        uses: docker/login-action@v3.3.0
+        if: ${{ github.event_name == 'release' }}
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v6.7.0
+        if: ${{ github.event_name == 'release' }}
```
Comment (ludeeus) [inline .github/workflows/release.yml]:
```suggestion
```

File: .github/workflows/release.yml (hunk 5 above)
Comment (klejejs) [reply .github/workflows/release.yml]: No, not right now. We can add it back If we decide to use Arm at some point.

**Review by ludeeus — APPROVED** [review]

**Pull request merged**